common/persistence/sql/storage/mysql/execution.go

```diff
@@ -0,0 +1,510 @@
+}
+
+// SelectFromCurrentExecutions reads one or more rows from current_executions table
+func (mdb *DB) SelectFromCurrentExecutions(filter *sqldb.CurrentExecutionsFilter) (*sqldb.CurrentExecutionsRow, error) {
+	var row sqldb.CurrentExecutionsRow
+	err := mdb.conn.Get(&row, getCurrentExecutionQry, filter.ShardID, filter.DomainID, filter.WorkflowID)
+	return &row, err
+}
+
+// DeleteFromCurrentExecutions deletes a single row in current_executions table
+func (mdb *DB) DeleteFromCurrentExecutions(filter *sqldb.CurrentExecutionsFilter) (sql.Result, error) {
+	return mdb.conn.Exec(deleteCurrentExecutionQry, filter.ShardID, filter.DomainID, filter.WorkflowID)
+}
+
+// LockCurrentExecutions acquires a write lock on a single row in current_executions table
+func (mdb *DB) LockCurrentExecutions(filter *sqldb.CurrentExecutionsFilter) (string, error) {
+	var runID string
+	err := mdb.conn.Get(&runID, lockCurrentExecutionQry, filter.ShardID, filter.DomainID, filter.WorkflowID)
+	return runID, err
+}
+
+// LockCurrentExecutionsJoinExecutions joins a row in current_executions with executions table and acquires a
+// write lock on the result
+func (mdb *DB) LockCurrentExecutionsJoinExecutions(filter *sqldb.CurrentExecutionsFilter) ([]sqldb.CurrentExecutionsRow, error) {
+	var rows []sqldb.CurrentExecutionsRow
+	err := mdb.conn.Select(&rows, getCurrentExecutionQryForUpdate, filter.ShardID, filter.DomainID, filter.WorkflowID)
+	return rows, err
+}
+
+// InsertIntoTransferTasks inserts one or more rows into transfer_tasks table
+func (mdb *DB) InsertIntoTransferTasks(rows []sqldb.TransferTasksRow) (sql.Result, error) {
+	return mdb.conn.NamedExec(createTransferTasksQry, rows)
+}
+
+// SelectFromTransferTasks reads one or more rows from transfer_tasks table
+func (mdb *DB) SelectFromTransferTasks(filter *sqldb.TransferTasksFilter) ([]sqldb.TransferTasksRow, error) {
+	var rows []sqldb.TransferTasksRow
+	err := mdb.conn.Select(&rows, getTransferTasksQry, filter.ShardID, *filter.MinTaskID, filter.MaxTaskID)
+	return rows, err
+}
+
+// DeleteFromTransferTasks deletes one or more rows from transfer_tasks table
+func (mdb *DB) DeleteFromTransferTasks(filter *sqldb.TransferTasksFilter) (sql.Result, error) {
+	if filter.MinTaskID != nil {
+		return mdb.conn.Exec(rangeDeleteTransferTaskQry, filter.ShardID, *filter.MinTaskID, *filter.MaxTaskID)
+	}
+	return mdb.conn.Exec(deleteTransferTaskQry, filter.ShardID, *filter.TaskID)
+}
+
+// InsertIntoTimerTasks inserts one or more rows into timer_tasks table
+func (mdb *DB) InsertIntoTimerTasks(rows []sqldb.TimerTasksRow) (sql.Result, error) {
+	return mdb.conn.NamedExec(createTimerTasksQry, rows)
+}
+
+// SelectFromTimerTasks reads one or more rows from timer_tasks table
+func (mdb *DB) SelectFromTimerTasks(filter *sqldb.TimerTasksFilter) ([]sqldb.TimerTasksRow, error) {
+	var rows []sqldb.TimerTasksRow
+	err := mdb.conn.Select(&rows, getTimerTasksQry, filter.ShardID, *filter.MinVisibilityTimestamp,
```

Comment:
Could we document in the filter structs or in DB interface methods when pointer fields are required?
Looking at the API it is not clear that it is going to dereference these pointers without checking.